src/main/java/bio/terra/testrunner/common/utils/KubernetesClientUtils.java

```diff
@@ -285,6 +297,50 @@ public static void buildKubernetesClientObjectWithClientKey(
 
     kubernetesClientCoreObject = new CoreV1Api();
     kubernetesClientAppsObject = new AppsV1Api();
+
+    // Import MCTerra Component versions from ConfigMap and outputs JSON-ready format
+    importComponentVersions();
+  }
+
+  /**
+   * This method imports MCTerra Component versions deployed to the target Kubernetes namespace.
+   *
+   * <p>A ConfigMap is used to store the terra-helmfile versions manifest which serve as the ground
+   * truth of MCTerra Component versions for all environments.
+   *
+   * <p>In order to import MCTerra Component versions from the Config Map, the Test Runner
+   * Kubernetes SA must be granted the following RBAC Role.
+   *
+   * <p>- apiGroups: [""] resources: ["configmaps"] resourceNames: ["terra-component-version"]
+   * verbs: ["get", "patch", "update"] . * For more information of RBAC Roles, please refer to the
+   * following documentation https://kubernetes.io/docs/reference/access-authn-authz/rbac/
+   *
+   * <p>Note: The name of the Component Version Identification ConfigMap: terra-component-version is
+   * built in to this process.
+   */
+  private static void importComponentVersions() {
+    try {
+      V1ConfigMap config =
+          getKubernetesClientCoreObject()
+              .readNamespacedConfigMap("terra-component-version", namespace, null, null, null);
+      Map<String, String> configMap = config.getData();
+      componentVersions =
+          configMap.entrySet().stream()
+              .collect(
+                  Collectors.toMap(
+                      entry -> entry.getKey().replace(".properties", ""),
+                      entry -> // A unique state of MCTerra deployment can employ multiline
+                          // key=value properties.
+                          // The purpose of this code is to collect all these properties as a map
+                          // for each and every MCTerra Component and return a map of map structure.
+                          Arrays.stream(entry.getValue().split("\\R"))
+                              .collect(
+                                  Collectors.toMap(
+                                      versionCfg -> versionCfg.split("=")[0],
+                                      versionCfg -> versionCfg.split("=")[1]))));
+    } catch (ApiException e) {
+      logger.debug(e.getResponseBody());
+    }
```

Comment:
I think most other methods in this class throw ApiException -- is there a reason this method should be different?

Reply:
I made this method throw an exception. Also made this method public and get rid of the private static variable `componentVersions`. The TestRunner now manages rendering the component versions.